team/bundles/org.eclipse.compare/compare/org/eclipse/compare/contentmergeviewer/TextMergeViewer.java

```diff
@@ -5421,8 +5421,8 @@ private void updateToolbarLabel() {
 		int differenceCount = fMerger.changesCount();
 		if (tbm != null) {
 
-			String label = differenceCount > 1 ? differenceCount + " Differences" //$NON-NLS-1$
-					: differenceCount == 1 ? differenceCount + " Difference" : "No Difference"; //$NON-NLS-1$ //$NON-NLS-2$
+			String label = differenceCount > 1 ? differenceCount + CompareMessages.TextMergeViewer_differences
+					: differenceCount == 1 ? differenceCount + CompareMessages.TextMergeViewer_difference : CompareMessages.TextMergeViewer_no_difference; //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
This is still not good for localization. There are languages in which the plural for two items is different from the plural of > 2 items. Just take a look at https://lingohub.com/blog/pluralization#cldr-overview . Pluralization is hard.

A better way to do this is to leave the choice to the translator. Using `MessageFormat.format` you could define a single message
```
TextMergeViewer_differences={0,choice,0#No differences|1#1 difference|1<{0} differences}
````
and then
```
String label = MessageFormat.format(CompareMessages.TextMergeViewer_differences, differenceCount);
```
This gives translators much more leeway in producing correct translations. It'll also format the number according to localized rules, for instance with thousands separators.

Reply:
I didn't know either that this was possible. Nice
@lathapatil: Can you adapt your PR accordingly?

Reply:
Sure, I will check on this.

Reply:
It is possible, though standard Java MessageFormat still is not powerful enough to handle all cases. A translator would still struggle to get the right plurals for most slavic languages. ICU has full support for doing this right, but unfortunately Eclipse ditched ICU a while ago "because it was too big".

Reply:
The main reason why we ditched ICU was that maintenance of this library was not actively down. IIRC there was one part time person working on it in IBM with his focus on other things.